### ISSUE TITLE
decode an id into controller

### ISSUE THREAD

**Issue by physio** [sschlein/obscure#3]
Good morning, sorry for my english.

i used your package into laravel 5.3 into api application.
in routes i use:
`Route::Resource('example', '\App\Api\V1\Controllers\ExampleController';`

in my controller into index() method i use Obscure::encode($example->id) and in output i can see the id obscured. this is good for me. i thik that all my configuration it is ok.

i don't undestand as i must to work for save (or delete) an model. how i can decode the $id? can i continue to use Route::resource?

reguards

**Comment by sschlein** [comment]
Hi,

honestly, I haven't tested it in a 5.3 application. There are some downsides when you want to use route model binding because there is a middleware for your routes which automatically decodes $id parameters in routes. This is also the reason I stopped working on the package because I did not find a way to make it work with route model binding.

If you route to show a blog post  is `/post/{id}`, your method would look like:

```
public function show($id)
{
    return view('post.show', ['post' => Post::find($id)]);
}
```

Even if your URL shows a hash, the id in the method is already decoded.
